app/views/two_factor_authentication/otp_verification/show.html.slim

```diff
@@ -15,7 +15,7 @@ p.mt-tiny.mb0#code-instructs == t('instructions.2fa.confirm_code',
   .col-12.sm-col-5.mb4.sm-mb0.sm-mr-20p.inline-block
     = text_field_tag(:code, '', value: @code_value, required: true,
       autofocus: true, pattern: '[0-9]*', class: 'col-12 field monospace mfa',
-      'aria-describedby': 'code-instructs')
+      'aria-describedby': 'code-instructs', maxlength: 6)
```

Comment:
would it be possible to reference vars in these views rather than a number? I can imagine a scenario in which we change the length and then the HTML validations are inaccurate.

Just dug around to figure out where these numbers come from -- they are from Devise! You can ref them like this:

```
 Devise.direct_otp_length
```

the *direct* otp is the code sent to the user. The other one is the one from authenticator apps. At least that is what I am thinking based on https://github.com/Houdini/two_factor_authentication